packages/neon_framework/lib/src/utils/request_manager.dart

```diff
@@ -26,7 +28,10 @@ typedef SerializeCallback<T> = String Function(T);
 
 /// A callback to revive cached values [String] into their original type [T].
 @internal
-typedef DeserializeCallback<T> = T Function(String);
+typedef DeserializeCallback<T> = FutureOr<T> Function(String);
+
+/// A callback to get the ETag and Expires of the resource.
+typedef GetCacheParameters = FutureOr<({String? etag, String? expires})?> Function();
```

Comment:
Can we create a separate class with these two parameters?
The class should also have a getter `isExpired` wisch automatically checks the parameter against the current DateTime.

The typedef could be removed in favor of an `AsyncValueGetter<CacheParameters>`

Reply:
It should also handle parsing the headers